sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/BaseAuthResolverHandler.cs

```diff
@@ -283,6 +283,133 @@ protected static List<IAuthSchemeOption> RetrieveSchemesFromEndpoint(Endpoint en
+            string preferenceString = ResolveAuthSchemePreference(clientConfig);
+            
+            if (string.IsNullOrWhiteSpace(preferenceString))
+            {
+                return authOptions;
+            }
+
+            try
+            {
+                // Parse the comma-separated preference list
+                var preferences = preferenceString.Split(',')
+                    .Select(s => s.Trim())
+                    .Where(s => !string.IsNullOrWhiteSpace(s))
+                    .ToList();
+
+                if (preferences.Count == 0)
+                {
+                    return authOptions;
+                }
+
+                var reordered = new List<IAuthSchemeOption>();
+                var remaining = new List<IAuthSchemeOption>(authOptions);
+
+                // Add schemes in preference order
+                foreach (var preferredName in preferences)
+                {
+                    var match = remaining.FirstOrDefault(opt =>
+                    {
+                        var shortName = AuthSchemeOption.GetNameFromSchemeId(opt.SchemeId);
+                        return string.Equals(shortName, preferredName, StringComparison.Ordinal);
+                    });
+                    
+                    if (match != null)
+                    {
+                        reordered.Add(match);
+                        remaining.Remove(match);
+                        Logger.DebugFormat("Applied preference for scheme: {0}", match.SchemeId);
+                    }
+                }
+
+                // Add any remaining schemes not in the preference list
+                foreach (var scheme in remaining)
+                {
+                    reordered.Add(scheme);
+                    Logger.DebugFormat("Added non-preferred scheme: {0}", scheme.SchemeId);
+                }
+
+                return reordered;
+            }
+            catch (ArgumentException ex)
+            {
+                Logger.Error(ex, "Failed to apply auth scheme preferences due to invalid argument, falling back to service-defined order");
+                return authOptions;
+            }
+            catch (InvalidOperationException ex)
+            {
+                Logger.Error(ex, "Failed to apply auth scheme preferences due to invalid operation, falling back to service-defined order");
+                return authOptions;
+            }
+        }
+
+        /// <summary>
+        /// Resolves authentication scheme preference using configuration precedence hierarchy.
+        /// </summary>
+        private string ResolveAuthSchemePreference(IClientConfig clientConfig)
+        {
+            // Check for legacy SignatureMethod configuration (backwards compatibility for S3)
+            if (clientConfig?.SignatureMethod != null && clientConfig.IsSignatureMethodExplicitlySet)
+            {
+                bool isS3Service = clientConfig.ServiceId?.Equals("S3", StringComparison.OrdinalIgnoreCase) == true ||
```

Comment:
(I left the other comment first...) I see the SEP mentions S3 (and Control) but I don't think this logic is needed here (the SEP was referring how the service model - which the auth resolvers are generated from - would keep older properties with the same values for backwards compatibility).